crates/astria-conductor/src/sequencer/mod.rs

```diff
@@ -285,22 +289,58 @@ async fn subscribe_new_blocks(
     .map_err(|_| ResubscriptionError::BackoffFailed)
 }
 
-fn forward_pending_block(
+fn forward_pending_block_or_resync(
+    expected_height: u32,
     executor_tx: executor::Sender,
+    pool: Pool<ClientProvider>,
+    pending_blocks: &mut FuturesOrdered<Ready<Result<SequencerBlockData, NewBlockStreamError>>>,
+    sync: &mut future::Fuse<Pin<Box<dyn Future<Output = eyre::Result<()>> + Send>>>,
     res: Result<SequencerBlockData, NewBlockStreamError>,
-) -> eyre::Result<()> {
+) -> eyre::Result<u32> {
+    use futures::FutureExt as _;
     let block = match res {
         Err(e) => {
             warn!(error.message = %e, error.cause = ?e, "response from sequencer block subscription was bad; dropping it");
-            return Ok(());
+            return Ok(expected_height);
         }
         Ok(block) => block,
     };
-    executor_tx
-        .send(ExecutorCommand::FromSequencer {
-            block: Box::new(block),
-        })
-        .wrap_err("failed sending new block received from sequencer to executor")
+    let block_height = block
+        .header()
+        .height
+        .value()
+        .try_into()
+        .wrap_err("failed converting cometbft to u32, although this should always work")?;
+
+    match expected_height.cmp(&block_height) {
+        // received block is at expected height: send to the executor
```

Comment:
what is the meaning of expected height here? is it the target height we should be syncing to?